src/helics/application_api/ValueFederate.cpp

```diff
@@ -264,14 +264,22 @@ void ValueFederate::registerValueInterfacesJson(const std::string& jsonString)
     if (doc.isMember("subscriptions")) {
         auto subs = doc["subscriptions"];
         for (const auto& sub : subs) {
+            bool skipNameTarget{false};
             auto name = fileops::getName(sub);
+            if (name.empty()) {
```

Comment:
What happens if name is not empty? I tried testing this also by setting a name in the subscriptions. Things still seemed to work pub/sub wise, but helicsSubscriptionGetTarget(ipt) returned what I put in the name key and not what I put in the target key.

Reply:
For subscriptions the key is the target,  so if you specify a name or a key that is the first target.  anything in target(s) field gets added to it.  

So unless you intentionally want to subscribe to multiple targets you really should only specify a key or a target.